### PR TITLE
CE-3480 Create a simple API to collect users feedback

### PR DESCRIPTION
**Pull request by adamkarminski**
This API is created to collect user's feedback provided in Mobile Experiments run by the ComEng  team.

@Wikia/community-engineering

### CODE REVIEW

File: includes/wikia/api/UserFeedbackStorageApiController.class.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	 * @requestParam int experimentId
+	 * @requestParam int variationId
+	 * @requestParam int userId
+	 * @requestParam string feedback
+	 * @requestParam int feedback_impression_count
+	 * @requestParam int feedback_previous_count
+	 * @throws MissingParameterApiException
+	 */
+	public function saveUserFeedback() {
+		$this->checkWriteRequest();
```
Comment (lukaszkonieczny) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php:22]: Does it work if user is an anon?

File: includes/wikia/api/UserFeedbackStorageApiController.class.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			if ( $paramValue === 0 ) {
+				throw new MissingParameterApiException( $paramName );
+			} else {
+				$requestData[$dbFieldName] = $paramValue;
+			}
+		}
+
+		$requestData = $requestData + [
+			'user_id' => $request->getInt( 'userId' ),
+			'feedback' => $request->getVal( 'feedback', '' ),
```
Comment (lukaszkonieczny) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: Please validate/escape this value

File: includes/wikia/api/UserFeedbackStorageApiController.class.php (hunk 1 above)
Comment (adamkarminski) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php:22]: Yup :)

File: includes/wikia/api/UserFeedbackStorageApiController.class.php (hunk 2 above)
Comment (adamkarminski) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: `$db->insert()` escapes values using the `addQuotes()` method. @macbre approves :)

Comment (macbre) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: I do agree with quote of me :)

File: includes/wikia/api/UserFeedbackStorageApiController.class.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	 * @throws MissingParameterApiException
+	 */
+	public function saveUserFeedback() {
+		$this->checkWriteRequest();
+		$request = $this->getRequest();
+
+		$requestParams = $this->getRequestData( $request );
+
+		$db = $this->getDatabaseForWrite();
+		$status = $db->insert( self::FEEDBACK_TABLE_NAME, $requestParams );
```
Comment (macbre) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: Please add `__METHOD__` as a third argument

File: includes/wikia/api/UserFeedbackStorageApiController.class.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	 * @requestParam string feedback
+	 * @requestParam int feedback_impression_count
+	 * @requestParam int feedback_previous_count
+	 * @throws MissingParameterApiException
+	 */
+	public function saveUserFeedback() {
+		$this->checkWriteRequest();
+		$request = $this->getRequest();
+
+		$requestParams = $this->getRequestData( $request );
```
Comment (macbre) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: So `data` or `params`? :)

Comment (adamkarminski) [inline includes/wikia/api/UserFeedbackStorageApiController.class.php]: Haha, true ;)

**Comment by macbre** [comment]
LGTM

**Comment by lukaszkonieczny** [comment]
LGTM!

**Pull request merged**